vendor/bat-native-ledger/src/bat/ledger/internal/properties/ballot_properties.cc

```diff
@@ -0,0 +1,39 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ledger/internal/properties/ballot_properties.h"
+
+namespace ledger {
+
+BallotProperties::BallotProperties() = default;
+
+BallotProperties::BallotProperties(
+    const BallotProperties& properties) {
+  viewing_id = properties.viewing_id;
+  surveyor_id = properties.surveyor_id;
+  publisher = properties.publisher;
+  count = properties.count;
+  prepare_ballot = properties.prepare_ballot;
+  proof_ballot = properties.proof_ballot;
+}
```

Comment:
It looks like there is nothing special in this copy-constructor. Having it declared explicitly prevents generating of move operations.

Reply:
Complex class/struct needs an explicit out-of-line copy constructor as per Chromium style guide

Reply:
I don't agree. 
The document explains the style checked by Chromium Style Checker tool.
(https://www.chromium.org/developers/coding-style/chromium-style-checker-errors)
For classes/structs encapsulating other complex types the style requires to have explicit out-of-line constructor/copy-constructor/destructor only if it does non-trival initialization or copying. For other cases, including the one I mentioned above, we should stick with `defaulted` implementation. 

`Prefer to use =default`:
https://chromium.googlesource.com/chromium/src/+/master/styleguide/c++/c++-dos-and-donts.md#prefer-to-use

Also, `initialize members in the declaration where possible`:
https://chromium.googlesource.com/chromium/src/+/master/styleguide/c++/c++-dos-and-donts.md#initialize-members-in-the-declaration-where-possible